media/js/mozorg/technology-rust-video-test.js

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global YT */
+/* eslint no-unused-vars: [2, { "varsIgnorePattern": "onYouTubeIframeAPIReady" }] */
+
+// YouTube API hook has to be in global scope
+function onYouTubeIframeAPIReady() {
+    'use strict';
+
+    Mozilla.technologyOnYouTubeIframeAPIReady();
+}
+
+(function($) {
+
+    var tag = document.createElement('script');
+    tag.src = 'https://www.youtube.com/iframe_api';
+
+    var firstScriptTag = document.getElementsByTagName('script')[0];
+    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+
+    function onYouTubeIframeAPIReady() {
+        // show video modal when user clicks "watch now" button
+        $('.hero-rust-video a.video-play').attr('role', 'button').click(function(e) {
+            e.preventDefault();
+
+            var $videoContainer = $('#video-container');
+            var video = $videoContainer.find('.video')[0];
```

Comment:
A pretty easy performance improvement would be moving these two vars to just before `function onYouTubeIframeAPIRead()`. As is, the DOM is re-queried each time the user clicks the button (which admittedly will probably only be once per visit, but it's a simple copy pasta).